controllers/servingruntime_controller.go

```diff
@@ -222,21 +259,25 @@ func (r *ServingRuntimeReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		AnnotationsMap:      cfg.RuntimePodAnnotations,
 		LabelsMap:           cfg.RuntimePodLabels,
 	}
+	log.Info("==== ChinDebug sr reconcile 5 ==== before delete mmDeployment", "mmEnabled", mmEnabled)
 
 	// if the runtime is disabled, delete the deployment
-	if rt.Spec.IsDisabled() || !rt.Spec.IsMultiModelRuntime() || !mmEnabled {
+	if (*spec).IsDisabled() || !(*spec).IsMultiModelRuntime() || !mmEnabled {
 		log.Info("Runtime is disabled, incompatible with modelmesh, or namespace is not modelmesh-enabled")
 		if err = mmDeployment.Delete(ctx, r.Client); err != nil {
 			return ctrl.Result{}, fmt.Errorf("could not delete the model mesh deployment: %w", err)
 		}
 		return ctrl.Result{}, nil
 	}
+	log.Info("==== ChinDebug sr reconcile 6 ==== no need to delete mmDeployment", "mmEnabled", mmEnabled)
 
-	replicas, requeueDuration, err := r.determineReplicasAndRequeueDuration(ctx, log, cfg, rt)
+	replicas, requeueDuration, err := r.determineReplicasAndRequeueDuration(ctx, log, cfg, spec, rtName)
```

Comment:
```suggestion
	replicas, requeueDuration, err := r.determineReplicasAndRequeueDuration(ctx, log, cfg, spec, req.NamespacedName)
```
(no need for intermediate var)